variable/include/scipp/variable/variable_factory.h

```diff
@@ -19,37 +19,32 @@ class SCIPP_VARIABLE_EXPORT AbstractVariableMaker {
 public:
   virtual ~AbstractVariableMaker() = default;
   virtual bool is_bins() const = 0;
-  virtual Variable
-  create(const DType elem_dtype, const Dimensions &dims,
-         const units::Unit &unit, const bool variances,
-         const std::vector<VariableConstView> &parents) const = 0;
-  virtual Dim elem_dim(const VariableConstView &var) const = 0;
-  virtual DType elem_dtype(const VariableConstView &var) const = 0;
-  virtual units::Unit elem_unit(const VariableConstView &var) const = 0;
-  virtual void expect_can_set_elem_unit(const VariableView &var,
+  virtual Variable create(const DType elem_dtype, const Dimensions &dims,
+                          const units::Unit &unit, const bool variances,
+                          const std::vector<Variable> &parents) const = 0;
```

Comment:
Yes, I had a TODO in this which I removed. It will happen about once per `transform` call. Not sure how to best avoid it? We get the parents as a parameter pack, so we'd need a variable-length type that can hold references? Maybe using `std::reference_wrapper`?

Reply:
I think the reference wrapper would work. It is a bit awkward to use but if it it localised to a few internal cases, then it should be fine.